### PR TITLE
Shorten XML validation error message

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
Only the relevant XML errors are generated if a file doesn't conform to the schema (filename, line, column, and error)

- Adds XmlError class
- Consolidates functions in `unity.py`
- closes issue #30

### CODE REVIEW

File: w3d_translator/validate.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import shutil
+from lxml import etree
 from pathlib import Path
 
-from errors import ValidationError
+from errors import ValidationError, XmlError
+
+schema = etree.XMLSchema(file="schema/caveschema.xsd")
```
Comment (eldu) [inline w3d_translator/validate.py:7]: Is this a global variable that doesn't change after this initialization? If so, I'd probably would make it all caps. `SCHEMA`

File: w3d_translator/w3d_translator.py [hunk 2]
```diff
@@ -46,59 +39,42 @@ def farewell():
     exit(0)
 
 
-# Farewell message on error
-def farewell_error():
+# Run doctests
+def run_tests():
+    typer.echo("Running Tests")
```
Comment (eldu) [inline w3d_translator/w3d_translator.py:44]: OOh a fun thing to do with tests is to also time them. If you're interested in optimizing and making things faster. No worries if not. I figured I'd toss it out there.

https://docs.python.org/3/library/timeit.html

File: w3d_translator/w3d_translator.py [hunk 3]
```diff
@@ -46,59 +39,42 @@ def farewell():
...
                 typer.echo(red(e), err=True)
-                typer.echo(red(f"Skipping {file.name}"), err=True)
             else:
-                # Build Story dataclass and Unity project
                 story = generateDS.parse(file, silence=True)
-                print(story.member_data_items_.keys())
                 build_project(unity_dir, story)
-
-    except (ValidationError, UnityError, TranslationError) as e:
+    except (ValidationError, UnityError) as e:
```
Comment (eldu) [inline w3d_translator/w3d_translator.py:77]: is there any chance that there are other errors? do we need another generic except block?

File: w3d_translator/validate.py (hunk 1 above)
Comment (eldu) [reply w3d_translator/validate.py:7]: https://peps.python.org/pep-0008/#constants

**Review by eldu — APPROVED** [review]
Looks good to me Rob! I added in a few comments mostly questions.

File: w3d_translator/w3d_translator.py (hunk 3 above)
Comment (RobertGemmaJr) [reply w3d_translator/w3d_translator.py:77]: Nothing that shouldn't crash the program anyways I don't think - especially while I'm still developing things. I'd rather python just error out and debug from there than catch anything at this stage

File: w3d_translator/validate.py [hunk 4]
```diff
@@ -34,3 +37,21 @@ def validate_out(dir: Path, force: bool):
...
+        schema.assertValid(etree.parse(file))
+    except etree.DocumentInvalid as error:
+        # Display Validation errors
+        error_list = "\n".join(
+            [
+                f"{e.filename}:{e.line}:{e.column}:  {e.message}"
+                for e in error.error_log
+            ]
+        )
+        raise XmlError(file.name, error_list)
```
[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply w3d_translator/validate.py:54]: Just for readability. I'm essentially recreating `e.__repr__` with only the properties I care about

File: w3d_translator/validate.py (hunk 1 above)
Comment (RobertGemmaJr) [reply w3d_translator/validate.py]: Oh, yes. Good catch! I changed it

**Pull request merged**